### ISSUE TITLE
Test 셋으로 모델을 평가할 때 epoch, step을 어느정도 해야 적절한가요?

### ISSUE THREAD

**Issue by JunjaK** [KHU-Dcom-Study/2018-1-DeepLearning-Study#47]
## Detail
cat vs dog cnn에서 훈련시킨 모델을 불러들여 약 1000step을 돌려보았습니다. 처음 300step 정도만 loss를 찾는 과정이 있어 정확도가 많이 낮았지만 이후에는 거의 100%를 유지했습니다. step을 좀 더 늘인다면 더 높은 평균 정확도를 얻을 수 있을 텐데 여기서 step을 더 늘이는 것이 맞는 것인지 궁금합니다.
(training set 개, 고양이 각각 1만장씩, test set 개, 고양이 각각 2,500장씩 사용했습니다.)
(test셋을 원래 모델에 집어 넣어 정확도를 구하는 test 방법이  맞는 것인지도 궁금합니다. 애초에 test 방법이 잘못되어 있는 것일 수 있어서...)
![test](https://user-images.githubusercontent.com/30404630/40068891-ad9e7386-58a4-11e8-8b1b-a728e0957e04.png)
![test](https://user-images.githubusercontent.com/30404630/40068896-af310150-58a4-11e8-87e0-32e79ce48a01.png)

**Comment by kairos03** [comment]
트레이닝이 100%가 나온다고 좋은것은 아닙니다. 위와 같은 경우 오버피팅을 의심해 볼 수 있습니다. 
트레이닝 후 해당 모델에 test 를 돌리는것이 맞는데 코드를 봐야 알 수 있을것 같네요. 
오버피팅이 의심되는 경우 dropout을 사용하거나, early stop, 즉 epoch를 낮추는 것이 도움이 됩니다.
